packages/editor/src/components/post-template/block-theme.js

```diff
@@ -67,14 +67,21 @@ export default function BlockThemeControl( { id } ) {
 					<MenuGroup>
 						<MenuItem
 							onClick={ ( event ) => {
-								editTemplate.onClick( event );
+								editTemplate.loadEntity( event );
 								onClose();
 								createSuccessNotice(
 									__(
 										'Editing template. Changes made here affect all posts and pages that use the template.'
 									),
 									{
 										type: 'snackbar',
+										actions: [
+											{
+												label: __( 'Go back' ),
+												onClick: () =>
+													changeEntity.goBack(),
+											},
+										],
```

Comment:
Separating `hasHistory` from `goBack` allows us to add these `goBack` links to nofications prior to moving to the new entity.

Reply:
I think I'd need to see an example of it not working to debug it.

My take on it is:
- The function is a closure, so it should be able to access the latest value of `goBack` from the current scope.
- [The code in trunk always provides an `onClick` handler](https://github.com/WordPress/gutenberg/blob/beb10327bc8942d5e0a7572242846d8d49d67aac/packages/editor/src/components/post-template/block-theme.js#L73-L77), and doesn't make any assumptions about whether the user can go back, so it seems ok not to check `hasHistory`.

Reply:
Sorry, my explanation was not clear here. The issue is that currently in trunk we only define `goBack`[ if there is some history](https://github.com/WordPress/gutenberg/blob/trunk/packages/edit-post/src/hooks/use-post-history.js#L71). So currently when loading `/post-template/block-theme.js` `goBack` is undefined because there is no history.

We are doing this because we are also using `goBack` to indicate that there is some `history`, eg. [here ](https://github.com/WordPress/gutenberg/blob/trunk/packages/edit-post/src/components/header/index.js#L80), and then using that to dictate if the back link should appear, etc. So, in this current setup if we always returned `goBack` as defined to use it in the likes of the above notification actions then the back links, etc. appear in places we don't want/need them.

Combining `goBack` with `hasHistory` made sense in the limited use case of the pattern editing, but doesn't extend to the wider use cases, so the two need to be separated - but as you note below `hasHistory` probably isn't the right term to use.

Reply:
> So, in this current setup if we always return goBack as defined without checking history, then in the places where we are using the presence of goBack to indicate history it is always going to be true so we end up with the doc bar and back links in the initial post page, etc.

The part I don't understand is why `goBack` always needs to be defined?

Reply:
It needs to always be defined, because at [this point here](https://github.com/WordPress/gutenberg/pull/57745/files#diff-8ab52da5fbf16ef635a4720adc66b21a530f9a29649a14bb722e4eabe401ea8cR88) when the edit template modal is loaded in the edit post side panel you are still just editing a post, and there is no history, so currently `goBack` would not be defined so a `Go back` action can't be added to the success notice.  

The only way around this without having `goBack` always defined would be to trigger the success notice only after you have loaded the template in the editor - which doesn't seem like a good option and not sure how you would actually do that.